More info for zypper install phase

jayvdb opened the pull request.

Mixih commented on Dockerfile:
```diff
@@ -19,12 +19,12 @@ RUN zypper addrepo http://download.opensuse.org/repositories/home:illuusio/openS
       # luarocks
       --plus-repo http://download.opensuse.org/repositories/devel:languages:lua/openSUSE_Tumbleweed/ \
-      install \
+      install --details --download-in-advance \
```
perhaps you should explain the `--download-in-advance` in the commit?

jayvdb commented:
I've played around with all combinations of `-v` , `-vv` and `--details`.  Either `-v` or `--details` adds the list of all packages to be installed with precise information about where the (to be) *installed* package comes from, as data like:

(1)
```
R-KernSmooth                  
  2.23.15-11.1                  
  x86_64
  OSS                                                                                  
  openSUSE    
```

Either option also includes an entry for each *selected* package, indicating roughly the same thing in the form of:

(2)
```
Selecting 'libxml2-tools-2.9.4-4.1.x86_64' from repository 'OSS' for installation.
...
Selecting 'julia-0.6.0-56.8.x86_64' from repository 'http://download.opensuse.org/repositories/science/openSUSE_Tumbleweed/' for installation.
```

Using `-vv` seems to only add HTTP fetch information everywhere, which effectively provides similar information for each (to be) *installed* package in the form of:

(3)
```
Retrieving package perl-TimeDate-1.20-26.5.noarch (135/334),  45.5 KiB ( 93.1 KiB unpacked)
Retrieving: http://download.opensuse.org/tumbleweed/repo/oss/suse/noarch/perl-TimeDate-1.20-26.5.noarch.rpm [.done (6.9 KiB/s)]
```

Typically problems are caused by (to be) *installed* packages, which are dependencies of the *selected* packages.

Form (1) is the most sparse and resulting in the log always exceeding the size Travis naturally shows.

The most informative and *dense* information is form (3), but we cant have form (3) without also getting form (1). :(

As a result, I think we should have an ENV variable which can be enabled to add all of these flags, and they are disabled by default so that the logs normally fit within the Travis viewing size.

jayvdb committed:
Emit zypper verbose ouput only on error

Save the zypper verbose out, and emit it only on error,
and only emit the download links on success.

Closes https://github.com/coala/docker-coala-base/issues/49

jayvdb commented:
Here is what an intentionally broken install looks like , with all the lovely XML emitted into the log: https://travis-ci.org/jayvdb/docker-coala-base/builds/260167281

yukiisbored commented:
ack https://github.com/coala/docker-coala-base/pull/149/commits/18a46703bec157bf340c7cd1892ff1437b25467e

yukiisbored commented:
@rultor merge pls

rultor commented:
> @rultor merge pls

@yukiisbored OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/12600-320484470)

rultor commented:
> @rultor merge pls

@jayvdb @yukiisbored Oops, I failed. You can see the full log [here](http://www.rultor.com/t/12600-320484470) (spent 2min)

```
++ pwd
+ docker run -t --rm -v /tmp/rultor-HYz0:/main --env=fork=git@github.com:jayvdb/docker-coala-base.git --env=fork_branch=zypper-more-info --env=head=git@github.com:coala/docker-coala-base.git --env=head_branch=master --env=pull_id=149 '--env=pull_title=More info for zypper install phase' --env=author=yukiisbored '--env=scripts=( '\''export '\''\'\'''\''fork=git@github.com:jayvdb/docker-coala-base.git'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''fork_branch=zypper-more-info'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''head=git@github.com:coala/docker-coala-base.git'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''head_branch=master'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''pull_id=149'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''pull_title=More info for zypper install phase'\''\'\'''\'''\'' '\'';'\'' '\''export '\''\'\'''\''author=yukiisbored'\''\'\'''\'''\'' '\'';'\'' '\''bash .ci/check_maintainership.sh'\'' '\'';'\'' )' --hostname=docker --privileged --memory=6g --memory-swap=16g --oom-kill-disable --cidfile=/tmp/rultor-HYz0/cid -w=/main --name=coala_docker-coala-base_149 yegor256/rultor /main/entry.sh
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
+ set -e
+ set -o pipefail
+ shopt -s dotglob
+ useradd -m -G sudo r
+ echo '%sudo ALL=(ALL) NOPASSWD:ALL'
+ cp -R /root/.bashrc /root/.composer /root/.gem /root/.gnupg /root/.m2 /root/.profile /root/texmf /home/r
+ cp -R ./cid ./end.sh ./entry.sh ./pid ./repo ./rultor_github_secrets.sh ./run.sh ./script.sh ./stdout /home/r
+ rm -rf repo
+ chown -R r:r /home/r
+ chmod a+x /home/r/script.sh
+ su --login r --command /home/r/script.sh
+ set -e
+ set -o pipefail
+ shopt -s expand_aliases
+ alias 'sudo=sudo -i'
+ export HOME=/home/r
+ HOME=/home/r
+ cd /home/r/repo
+ export fork=git@github.com:jayvdb/docker-coala-base.git
+ fork=git@github.com:jayvdb/docker-coala-base.git
+ export fork_branch=zypper-more-info
+ fork_branch=zypper-more-info
+ export head=git@github.com:coala/docker-coala-base.git
+ head=git@github.com:coala/docker-coala-base.git
+ export head_branch=master
+ head_branch=master
+ export pull_id=149
+ pull_id=149
+ export 'pull_title=More info for zypper install phase'
+ pull_title='More info for zypper install phase'
+ export author=yukiisbored
+ author=yukiisbored
+ bash .ci/check_maintainership.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   117  100   117    0     0    274      0 --:--:-- --:--:-- --:--:--   274
{
  "state": "active",
  "role": "member",
  "url": "https://api.github.com/teams/1238757/memberships/yukiisbored"
}
+ mv /home/r/repo .
++ whoami
+ chown -R root repo
+ '[' -n '' ']'
++ whoami
+ sudo chown -R rultor repo
+ cd repo
+ git push origin master
remote: error: GH006: Protected branch update failed for refs/heads/master.        
remote: error: Required status check "continuous-integration/travis-ci/pr" is failing.        
To git@github.com:coala/docker-coala-base.git
 ! [remote rejected] master -> master (protected branch hook declined)
error: failed to push some refs to 'git@github.com:coala/docker-coala-base.git'
container 05d29f6693ae4b2d961821d501bd603b44883a5c796f91179abcd4291c36bc79 is dead
Sun Aug  6 06:12:02 CEST 2017

```

yukiisbored commented:
@rultor merge --do-not-fail

rultor commented:
> @rultor merge --do-not-fail

@yukiisbored OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/12600-320485017)

rultor commented:
> @rultor merge --do-not-fail

@yukiisbored Done! FYI, the full log is [here](http://www.rultor.com/t/12600-320485017) (took me 2min)
